monitoring/promexp/main.go

```diff
@@ -58,6 +77,36 @@ func main() {
 </body></html>`))
 	})
 
+	// Forces a data push.
+	http.HandleFunc("/push", func(w http.ResponseWriter, r *http.Request) {
+		var msg string
+		if metrics, err := exporter.CollectRaw(); err == nil {
+			b := new(bytes.Buffer)
+			ts := time.Now().UnixNano()
+			for k, v := range metrics {
+				fmt.Fprintf(b, "%s value=%f %d\n", k, v, ts)
+			}
+			reqBody := b.String()
+			if req, err := http.NewRequest("POST", targetAddress, bytes.NewReader([]byte(reqBody))); err == nil {
+				req.Header.Add("Authorization", tokenHeader)
+				if resp, err := http.DefaultClient.Do(req); err != nil {
+					msg = "Fail: " + err.Error()
+				} else {
+					msg = "ok - " + resp.Status
```

Comment:
I think response body `resp` should be closed.

Reply:
Good catch. Done.